api/rest/client/client_test.go

```diff
@@ -63,249 +61,54 @@ func TestMultiaddressTakesPreference(t *testing.T) {
 	if c.urlPrefix != "http://127.0.0.1:9094" {
 		t.Error("default should be used")
 	}
+}
 
-	cfg = &Config{
+func TestMultiaddressPreference(t *testing.T) {
+	addr, _ := ma.NewMultiaddr(apiAddr)
+	cfg := &Config{
 		APIAddr:           addr,
 		Host:              "localhost",
 		Port:              "9094",
 		DisableKeepAlives: true,
 	}
-	c, err = NewClient(cfg)
+	c, err := NewClient(cfg)
 	if err != nil {
 		t.Fatal(err)
 	}
 	if c.urlPrefix != "http://127.0.0.1:10005" {
 		t.Error("APIAddr should be used")
 	}
+}
 
-	cfg = &Config{
+func TestHostPort(t *testing.T) {
+	cfg := &Config{
 		APIAddr:           nil,
 		Host:              "localhost",
 		Port:              "9094",
 		DisableKeepAlives: true,
 	}
-	c, err = NewClient(cfg)
+	c, err := NewClient(cfg)
 	if err != nil {
 		t.Fatal(err)
 	}
 	if c.urlPrefix != "http://localhost:9094" {
 		t.Error("Host Port should be used")
 	}
+}
 
+func TestDNSMultiaddress(t *testing.T) {
 	addr2, _ := ma.NewMultiaddr("/dns4/localhost/tcp/1234")
-	cfg = &Config{
+	cfg := &Config{
 		APIAddr:           addr2,
 		Host:              "localhost",
 		Port:              "9094",
 		DisableKeepAlives: true,
 	}
-	c, err = NewClient(cfg)
+	c, err := NewClient(cfg)
 	if err != nil {
 		t.Fatal(err)
 	}
 	if c.urlPrefix != "http://127.0.0.1:1234" {
 		t.Error("bad resolved address")
 	}
 }
-
-func TestVersion(t *testing.T) {
```

Comment:
Why did these tests make codeclimate unhappy?  Are we planning to replace them?

Reply:
oh, i forgot to add the file, i moved them